<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
<!-- 2020-02-03 Po 07:49 -->
<meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>RepeatExplorer documentation</title>
<meta name="generator" content="Org mode" />
<meta name="author" content="Petr Novak" />
<style type="text/css">
 <!--/*--><![CDATA[/*><!--*/
  .title  { text-align: center;
             margin-bottom: .2em; }
  .subtitle { text-align: center;
              font-size: medium;
              font-weight: bold;
              margin-top:0; }
  .todo   { font-family: monospace; color: red; }
  .done   { font-family: monospace; color: green; }
  .priority { font-family: monospace; color: orange; }
  .tag    { background-color: #eee; font-family: monospace;
            padding: 2px; font-size: 80%; font-weight: normal; }
  .timestamp { color: #bebebe; }
  .timestamp-kwd { color: #5f9ea0; }
  .org-right  { margin-left: auto; margin-right: 0px;  text-align: right; }
  .org-left   { margin-left: 0px;  margin-right: auto; text-align: left; }
  .org-center { margin-left: auto; margin-right: auto; text-align: center; }
  .underline { text-decoration: underline; }
  #postamble p, #preamble p { font-size: 90%; margin: .2em; }
  p.verse { margin-left: 3%; }
  pre {
    border: 1px solid #ccc;
    box-shadow: 3px 3px 3px #eee;
    padding: 8pt;
    font-family: monospace;
    overflow: auto;
    margin: 1.2em;
  }
  pre.src {
    position: relative;
    overflow: visible;
    padding-top: 1.2em;
  }
  pre.src:before {
    display: none;
    position: absolute;
    background-color: white;
    top: -10px;
    right: 10px;
    padding: 3px;
    border: 1px solid black;
  }
  pre.src:hover:before { display: inline;}
  /* Languages per Org manual */
  pre.src-asymptote:before { content: 'Asymptote'; }
  pre.src-awk:before { content: 'Awk'; }
  pre.src-C:before { content: 'C'; }
  /* pre.src-C++ doesn't work in CSS */
  pre.src-clojure:before { content: 'Clojure'; }
  pre.src-css:before { content: 'CSS'; }
  pre.src-D:before { content: 'D'; }
  pre.src-ditaa:before { content: 'ditaa'; }
  pre.src-dot:before { content: 'Graphviz'; }
  pre.src-calc:before { content: 'Emacs Calc'; }
  pre.src-emacs-lisp:before { content: 'Emacs Lisp'; }
  pre.src-fortran:before { content: 'Fortran'; }
  pre.src-gnuplot:before { content: 'gnuplot'; }
  pre.src-haskell:before { content: 'Haskell'; }
  pre.src-hledger:before { content: 'hledger'; }
  pre.src-java:before { content: 'Java'; }
  pre.src-js:before { content: 'Javascript'; }
  pre.src-latex:before { content: 'LaTeX'; }
  pre.src-ledger:before { content: 'Ledger'; }
  pre.src-lisp:before { content: 'Lisp'; }
  pre.src-lilypond:before { content: 'Lilypond'; }
  pre.src-lua:before { content: 'Lua'; }
  pre.src-matlab:before { content: 'MATLAB'; }
  pre.src-mscgen:before { content: 'Mscgen'; }
  pre.src-ocaml:before { content: 'Objective Caml'; }
  pre.src-octave:before { content: 'Octave'; }
  pre.src-org:before { content: 'Org mode'; }
  pre.src-oz:before { content: 'OZ'; }
  pre.src-plantuml:before { content: 'Plantuml'; }
  pre.src-processing:before { content: 'Processing.js'; }
  pre.src-python:before { content: 'Python'; }
  pre.src-R:before { content: 'R'; }
  pre.src-ruby:before { content: 'Ruby'; }
  pre.src-sass:before { content: 'Sass'; }
  pre.src-scheme:before { content: 'Scheme'; }
  pre.src-screen:before { content: 'Gnu Screen'; }
  pre.src-sed:before { content: 'Sed'; }
  pre.src-sh:before { content: 'shell'; }
  pre.src-sql:before { content: 'SQL'; }
  pre.src-sqlite:before { content: 'SQLite'; }
  /* additional languages in org.el's org-babel-load-languages alist */
  pre.src-forth:before { content: 'Forth'; }
  pre.src-io:before { content: 'IO'; }
  pre.src-J:before { content: 'J'; }
  pre.src-makefile:before { content: 'Makefile'; }
  pre.src-maxima:before { content: 'Maxima'; }
  pre.src-perl:before { content: 'Perl'; }
  pre.src-picolisp:before { content: 'Pico Lisp'; }
  pre.src-scala:before { content: 'Scala'; }
  pre.src-shell:before { content: 'Shell Script'; }
  pre.src-ebnf2ps:before { content: 'ebfn2ps'; }
  /* additional language identifiers per "defun org-babel-execute"
       in ob-*.el */
  pre.src-cpp:before  { content: 'C++'; }
  pre.src-abc:before  { content: 'ABC'; }
  pre.src-coq:before  { content: 'Coq'; }
  pre.src-groovy:before  { content: 'Groovy'; }
  /* additional language identifiers from org-babel-shell-names in
     ob-shell.el: ob-shell is the only babel language using a lambda to put
     the execution function name together. */
  pre.src-bash:before  { content: 'bash'; }
  pre.src-csh:before  { content: 'csh'; }
  pre.src-ash:before  { content: 'ash'; }
  pre.src-dash:before  { content: 'dash'; }
  pre.src-ksh:before  { content: 'ksh'; }
  pre.src-mksh:before  { content: 'mksh'; }
  pre.src-posh:before  { content: 'posh'; }
  /* Additional Emacs modes also supported by the LaTeX listings package */
  pre.src-ada:before { content: 'Ada'; }
  pre.src-asm:before { content: 'Assembler'; }
  pre.src-caml:before { content: 'Caml'; }
  pre.src-delphi:before { content: 'Delphi'; }
  pre.src-html:before { content: 'HTML'; }
  pre.src-idl:before { content: 'IDL'; }
  pre.src-mercury:before { content: 'Mercury'; }
  pre.src-metapost:before { content: 'MetaPost'; }
  pre.src-modula-2:before { content: 'Modula-2'; }
  pre.src-pascal:before { content: 'Pascal'; }
  pre.src-ps:before { content: 'PostScript'; }
  pre.src-prolog:before { content: 'Prolog'; }
  pre.src-simula:before { content: 'Simula'; }
  pre.src-tcl:before { content: 'tcl'; }
  pre.src-tex:before { content: 'TeX'; }
  pre.src-plain-tex:before { content: 'Plain TeX'; }
  pre.src-verilog:before { content: 'Verilog'; }
  pre.src-vhdl:before { content: 'VHDL'; }
  pre.src-xml:before { content: 'XML'; }
  pre.src-nxml:before { content: 'XML'; }
  /* add a generic configuration mode; LaTeX export needs an additional
     (add-to-list 'org-latex-listings-langs '(conf " ")) in .emacs */
  pre.src-conf:before { content: 'Configuration File'; }

  table { border-collapse:collapse; }
  caption.t-above { caption-side: top; }
  caption.t-bottom { caption-side: bottom; }
  td, th { vertical-align:top;  }
  th.org-right  { text-align: center;  }
  th.org-left   { text-align: center;   }
  th.org-center { text-align: center; }
  td.org-right  { text-align: right;  }
  td.org-left   { text-align: left;   }
  td.org-center { text-align: center; }
  dt { font-weight: bold; }
  .footpara { display: inline; }
  .footdef  { margin-bottom: 1em; }
  .figure { padding: 1em; }
  .figure p { text-align: center; }
  .equation-container {
    display: table;
    text-align: center;
    width: 100%;
  }
  .equation {
    vertical-align: middle;
  }
  .equation-label {
    display: table-cell;
    text-align: right;
    vertical-align: middle;
  }
  .inlinetask {
    padding: 10px;
    border: 2px solid gray;
    margin: 10px;
    background: #ffffcc;
  }
  #org-div-home-and-up
   { text-align: right; font-size: 70%; white-space: nowrap; }
  textarea { overflow-x: auto; }
  .linenr { font-size: smaller }
  .code-highlighted { background-color: #ffff00; }
  .org-info-js_info-navigation { border-style: none; }
  #org-info-js_console-label
    { font-size: 10px; font-weight: bold; white-space: nowrap; }
  .org-info-js_search-highlight
    { background-color: #ffff00; color: #000000; font-weight: bold; }
  .org-svg { width: 90%; }
  /*]]>*/-->
</style>
<link rel="stylesheet" type="text/css" href="style1.css" />
<script type="text/javascript">
/*
@licstart  The following is the entire license notice for the
JavaScript code in this tag.

Copyright (C) 2012-2019 Free Software Foundation, Inc.

The JavaScript code in this tag is free software: you can
redistribute it and/or modify it under the terms of the GNU
General Public License (GNU GPL) as published by the Free Software
Foundation, either version 3 of the License, or (at your option)
any later version.  The code is distributed WITHOUT ANY WARRANTY;
without even the implied warranty of MERCHANTABILITY or FITNESS
FOR A PARTICULAR PURPOSE.  See the GNU GPL for more details.

As additional permission under GNU GPL version 3 section 7, you
may distribute non-source (e.g., minimized or compacted) forms of
that code without the copy of the GNU GPL normally required by
section 4, provided you include this license notice and a URL
through which recipients can access the Corresponding Source.


@licend  The above is the entire license notice
for the JavaScript code in this tag.
*/
<!--/*--><![CDATA[/*><!--*/
 function CodeHighlightOn(elem, id)
 {
   var target = document.getElementById(id);
   if(null != target) {
     elem.cacheClassElem = elem.className;
     elem.cacheClassTarget = target.className;
     target.className = "code-highlighted";
     elem.className   = "code-highlighted";
   }
 }
 function CodeHighlightOff(elem, id)
 {
   var target = document.getElementById(id);
   if(elem.cacheClassElem)
     elem.className = elem.cacheClassElem;
   if(elem.cacheClassTarget)
     target.className = elem.cacheClassTarget;
 }
/*]]>*///-->
</script>
</head>
<body>
<div id="content">
<h1 class="title">RepeatExplorer documentation</h1>
<h1 id="clust"> Cluster annotation table </h1>

<dl class="org-dl">
<dt>Cluster</dt><dd>Cluster index, contains links to detailed cluster reports.</dd>
<dt>Supercluster</dt><dd>Supercluster index, contains links to detailed supercluster reports.</dd>
<dt>Proportion<code>[%]</code></dt><dd>Proportion of the reads in the cluster with respect to the amount of number of analyzed sequences.</dd>
<dt>Proportions adjusted<code>[%]</code></dt><dd>Adjusted genome proportion can differ from the Proportion<code>[%]</code> value if Perform automatic filtering of abundant satellite repeats was on. Sequences belonging to high-abundance satellites are partially removed from all-to-all comparison and clustering, causing the Proportion estimate for these satellites to be underestimated. Proportion adjusted provides a corrected estimate of genomic proportion for a particular satellite repeat.</dd>
<dt>Number of reads</dt><dd>Number of reads in the cluster.</dd>
<dt>Graph layout</dt><dd>Graph-based visualization of similarities among sequence reads.</dd>
<dt>Similarity hits</dt><dd>The proportion of reads in the clusters with similarity to REXdb or DNA reference databases. Only hits with proportion above 0.1% are shown.</dd>
<dt>Similarity hits to custom database</dt><dd>This column is optional and shows all similarity hits based on the search using BLASTN against a custom nucleotide database provided by the user.</dd>
<dt>LTR detection</dt><dd>This column shows whether an LTR with a primer binding site was detected in the contig assembly and the type of complementary tRNA.</dd>
<dt>Satellite probability</dt><dd>Empirical probability estimate that cluster sequences are derived from satellite repeat. This estimate is based on analysis of manually annotated clusters and experimentally validated satellite repeats.</dd>
<dt>TAREAN annotation</dt><dd>TAREAN divides clusters into five categories, high confidence satellites, low confidence satellites, putative LTR element, rDNA and other clusters where no potential tandem-like structure.</dd>
<dt>Consensus length</dt><dd></dd>

<dt>Consensus</dt><dd>Consensus sequence is the outcome of k‑mer–based analysis and represents the most probable satellite monomer sequence.</dd>
<dt>TAREAN k‑mer analysis</dt><dd>Hyperlink to individual clusters TAREAN k‑mer report.</dd>
<dt>Connected component index C</dt><dd>Proportion of nodes of the graph that are part of the largest strongly connected component.</dd>
<dt>Pair completeness index P</dt><dd>Proportion of reads in complete read pairs within the cluster.</dd>
<dt>TAREAN k‑mer coverage</dt><dd>Sum of relative frequencies of all k‑mers used for consensus sequence reconstruction.</dd>
<dt>|V|</dt><dd>Number of vertices of the graph.</dd>
<dt>|E|</dt><dd>Number of edges of the graph.</dd>
</dl>


<h1 id="superclust"> Supercluster annotation table </h1>

<dl class="org-dl">
<dt>Supercluster</dt><dd>Supercluster index.</dd>
<dt>Number of reads</dt><dd>Number of reads in supercluster.</dd>
<dt>Automatic annotation</dt><dd>Result of automatic supercluster annotation.</dd>
<dt>Similarity hits</dt><dd>Number of similarity hits against REXdb and DNA database are shown in the classification tree structure, together with the number of reads assigned to putative satellite cluster and information about detection of LTR/PBS. Parts of the tree without any evidence are pruned off.</dd>
<dt>TAREAN annotation</dt><dd>Clusters that are part of a supercluster and classified by TAREAN as putative satellites are listed here.</dd>
<dt>Clusters</dt><dd>Hyperlinked list of clusters that are part of superclusters.</dd>
</dl>

<h1 id="tra"> Tandem repeat analysis </h1>

<p>
TAREAN divides clusters into five categories with corresponding files in the archive:
</p>
<ul class="org-ul">
<li>High confidence satellites with consensus sequences in file TR<sub>consensus</sub><sub>rank</sub><sub>1</sub><sub>.fasta</sub></li>
<li>Low confidence satellites with consensus sequences in file TR<sub>consensus</sub><sub>rank</sub><sub>2</sub><sub>.fasta</sub></li>
<li>Putative LTR element with consensus sequences in file TR<sub>consensus</sub><sub>rank</sub><sub>3</sub><sub>.fasta</sub></li>
<li>rDNA with consensus in TR<sub>consensus</sub><sub>rank</sub><sub>4</sub><sub>.fasta</sub></li>
<li>Other clusters where no potential tandem-like structure was found</li>
</ul>

<p>
Summary tables from TAREAN html report include the following information: 
</p>

<dl class="org-dl">
<dt>Cluster</dt><dd>Cluster index, contains links to detailed cluster reports.</dd>
<dt>Proportion<code>[%]</code></dt><dd>Proportion of the reads in the cluster with respect to the amount of number of analyzed sequences.</dd>
<dt>Proportion adjusted<code>[%]</code></dt><dd>Adjusted genome proportion can differ from the Proportion<code>[%]</code> value if Perform automatic filtering of abundant satellite repeats was on. Sequences belonging to high-abundance satellites are partially removed from all-to-all comparison and clustering, causing the Proportion estimate for these satellites to be underestimated. Proportion adjusted provides a corrected estimate of genomic proportion for a particular satellite repeat.</dd>
<dt>Number of reads</dt><dd>Number of reads in the cluster</dd>
<dt>Satellite probability</dt><dd>Empirical probability estimate that cluster sequences are derived from satellite repeat. This estimate is based on analysis of manually annotated clusters and experimentally validated satellite repeats (Novák et al. 2017).⁠</dd>
<dt>Consensus length</dt><dd></dd>

<dt>Consensus</dt><dd>Consensus sequence is the outcome of k‑mer–based analysis and represents the most probable satellite monomer sequence.</dd>
<dt>Graph layout</dt><dd>Graph-based visualization of similarities among sequence reads.</dd>
<dt>TAREAN k‑mer analysis</dt><dd>Hyperlink to individual clusters TAREAN k‑mer report.</dd>
<dt>Connected component index C</dt><dd>Proportion of nodes of the graph that are part of the largest strongly connected component.</dd>
<dt>Pair completeness index P</dt><dd>Proportion of reads in complete read pairs within the cluster.</dd>
<dt>TAREAN k‑mer coverage</dt><dd>Sum of relative frequencies of all k‑mers used for consensus sequence reconstruction.</dd>
<dt>|V|</dt><dd>Number of vertices of the graph.</dd>
<dt>|E|</dt><dd>Number of edges of the graph.</dd>
<dt>PBS score</dt><dd>Primer binding site score, as detected by the TAREAN module. For clusters not analyzed by TAREAN, this value is not available.</dd>
<dt>Similarity hits</dt><dd>Similarity hits based on the search using BLASTN/BLASTX against built-in databases of known sequences. By default, this contains similarity hits to built-in databases that include rDNA, plastid, and mitochondrial sequences. If TAREAN was run within the RepeatExplorer2 pipeline, it will also contain information about similarity hits to the REXdb database.</dd>
<dt>Similarity hits to custom database</dt><dd>This column is optional and shows all similarity hits based on the search using BLASTN against a custom nucleotide database provided by the user.</dd>
</dl>
<p>
In individual clusters, the TAREAN report contains other variants of consensus sequences sorted by TAREAN k‑mer coverage score. For each consensus, the corresponding de Bruijn graph representation and corresponding sequence logo is shown.
</p>

<h1 id="kmer"> TAREAN k-mer analysis report </h1>

<p>
The TAREAN module generates k‑mer analysis report for each cluster assigned to a putative satellite, rDNA, or putative LTR category. Consensus sequences of these repeats are reconstructed using a k‑mer–based method using the most frequent k‑mers. Several k‑mer lengths are evaluated, and the best estimates of monomer consensus sequences are reported. K‑mer analysis summary contain the following  information:
</p>
<dl class="org-dl">
<dt>k‑mer length</dt><dd></dd>

<dt>Variant index</dt><dd>Each k‑mer of given length can yield multiple consensus variants. Variants are indexed.</dd>
<dt>k‑mer coverage score</dt><dd>The sum of proportions of all k‑mers used for reconstruction of particular monomer. If the value is 1, then all k‑mers from the corresponding cluster were used for reconstruction of the monomer, meaning that there is no variability. The more variable the monomer, the lower the k‑mer coverage score.</dd>
<dt>Consensus length</dt><dd></dd>

<dt>Consensus</dt><dd>Consensus sequence extracted from the position probability matrix.</dd>
<dt>k‑mer bases graph</dt><dd>Visualization of the de Bruijn graph. Each vertex correspond to single k‑mer. Size of vertex is proportional to the k‑mer frequency. The path that was used to reconstruct the monomer sequence is grayed out.</dd>
<dt>Sequence logo</dt><dd>Visualization of position probability matrix for the corresponding consensus variant.</dd>
</dl>
</div>
</body>
</html>
